_pages/dev/ssr/server-side-rendering-error-handling.md

```diff
@@ -0,0 +1,422 @@
+
+
+
+## Propagating errors to the server
+
+To propagate errors caught during server-side rendering to the ExpressJS server, the `CxErrorHandler` provided together with `Standardized SSR logging` feature has been extended. Now the error handler acts differently based on the platform it is running on thanks to multi error handlers:
+- `LoggingErrorHandler`, which logs the error using `LoggerService`;
+- `PropagatingToServerErrorHandler`, specially for SSR Error Handling purposes, which passes the error to the `CxCommonEngine` thanks to `PROPAGATE_ERROR_TO_SERVER` injection token. For more, see [CxCommonEngine](#cx-common-engine).
+
+To read more about Multi error handlers, see [Multi error handlers](#multi-error-handlers).
+
+**Note:** If you're providing your own error handler, extend the `CxErrorHandler` from Spartacus to ensure benefits from the SSR Error Handling.
+
+## CxCommonEngine
+
+`CxCommonEngine` is a core element of SSR Error Handling. It is a wrapper for Angular's [CommonEngine](https://github.com/angular/angular-cli/blob/e56adb062b86ecc538346412856bba57a8f378cf/packages/angular/ssr/src/common-engine.ts#L56) that allows to react on asynchronous errors that occurred during SSR. 
+The following is an implementation of the `CxCommonEngine`:
+```ts
+export class CxCommonEngine extends CommonEngine {
+  constructor(options?: CommonEngineOptions) {
+    super(options);
+  }
+
+  /**
+   * If an error is populated from the rendered applications
+   * (via `PROPAGATE_ERROR_TO_SERVER` callback), then such an error
+   * will be thrown and the result promise rejected - but only AFTER the rendering is complete.
+   * In other words, at first an error occurs and it's captured, then we wait until the rendering completes
+   * and ONLY then we reject the promise with the payload being the encountered error.
+   *
+   * Note: if more errors are captured during the rendering, only the first one will be used
+   *       as the payload of the rejected promise, others won't.
+   *
+   * @param {CommonEngineRenderOptions} options - The options to render.
+   * @returns {Promise<string>} Promise which resolves with the rendered HTML as a string
+   *                            OR rejects with the error, if any is propagated from the rendered app.
+   */
+  override async render(options: CommonEngineRenderOptions): Promise<string> {
+    let error: undefined | unknown;
+
+    return super
+      .render({
+        ...options,
+        providers: [
+          {
+            provide: PROPAGATE_ERROR_TO_SERVER,
+            useFactory: () => {
+              return (propagatedError: unknown) => {
+                // We're interested only the first propagated error
+                error ??= propagatedError;
+              };
+            },
+          },
+          ...(options.providers ?? []),
+        ],
+      })
+      .then((html: string) => {
+        if (error) {
+          throw error;
+        }
+        return html;
+      });
+  }
+}
+```
```

Comment:
Copy-pasted long pieces of our source code to the official docs may quickly become outdated.
What about placing here a link to our implementation? e.g.:
https://github.com/SAP/spartacus/blob/develop/core-libs/setup/ssr/engine/cx-common-engine.ts